code/modules/wiremod/components/gas_sensor.dm

```diff
@@ -0,0 +1,38 @@
+/**
+ * # Gas Sensor Component
+ *
+ * Return amount of certain gas in the air(in moles)
+ */
+/obj/item/circuit_component/gas_sensor
+	display_name = "Gas Sensor"
+
+	/// The input port
+	var/datum/port/input/input_port
+
+	/// The result from the output
+	var/datum/port/output/output
+
+	has_trigger = TRUE
+
+/obj/item/circuit_component/gas_sensor/Initialize()
+	. = ..()
+
+	input_port = add_input_port("Gas Name", PORT_TYPE_STRING)
+
+	output = add_output_port("Gas Amount", PORT_TYPE_NUMBER)
+
+/obj/item/circuit_component/gas_sensor/Destroy()
+	input_port = null
+	output = null
+	return ..()
+
+/obj/item/circuit_component/gas_sensor/input_received(datum/port/input/port)
+	. = ..()
+	if(.)
+		return
+
+	var/datum/gas_mixture/air = return_air()
+	var/gas_name = gas_name2path(input_port.input_value)
```

Comment:
This seems like a very bad way to go about this. Would it be possible to give them a dropdown of gas names instead of just hoping they'll remember the proper spelling of the one they want?

Reply:
not at present afaik, unfortunately

Reply:
You can do a basic dropdown if all you're looking at is the string.. Create a global list containing the list of gasses, then simply check the current_option to check which gas you are scanning for. This is how I do the dropdown for components like the typecheck and the arithmetic.

Reply:
![image](https://user-images.githubusercontent.com/37270891/120508082-52c8e700-c3bf-11eb-9dd2-69150f5eb40f.png)

Reply:
> 
> 
> You can do a basic dropdown if all you're looking at is the string. Create a global list containing the list of gasses, then simply check the current_option variable to check which gas you are scanning for. This is how I do the dropdown for components like the typecheck and the arithmetic.

Alright, please do this then @SmArtKar